sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/V1Table.scala

```diff
@@ -51,20 +52,18 @@ private[sql] case class V1Table(v1Table: CatalogTable) extends Table {
     }
   }
 
-  def catalogTable: CatalogTable = v1Table
-
-  lazy val options: Map[String, String] = {
-    v1Table.storage.locationUri match {
+  override lazy val properties: util.Map[String, String] = {
```

Comment:
If options are to be included in properties, I think that we should prefix them with `option.` so that we can recover them. I suggested this to Xin, who opened SPARK-28555 and was going to work on it, but I haven't seen a PR for it yet.

The idea is basically to be able to keep options and properties separate when passing through the v2 interface that only supports properties.

Since this PR updates V1Table to combine them, maybe we should include the change in this PR. What do you think?